src/components/Home/HomepageMap.tsx

```diff
@@ -32,21 +26,20 @@ export default function HomepageMap() {
     const minZoom = CONFIG.minZoom;
 
     // Create the state for the modal info
-    const [activeStation, setActiveStation] = useState(0);
-    const [markers, setMarkers] = useState<any>([]);
```

Comment:
To be honest I don't remember why had added this in the first place. Due to the flashing of the markers on each click?

Reply:
Yep, I believe that's right. Before the changes we had a bit too much logic in the provider. While exploring and reading more about the re-rendeing of the component in general, we found that when the state coming from the provider changes, it leads to the rendering of any of its children components. For us, it was re-rendering the whole component and that was leading to this flashing. We had found that using the state and memoizing the value, the flashing was resolved. However keeping the provider smaller with specific context I noticed we don't really need this, because indeed if there is a change in the stations, then we should correctly re-render.